text/0000-Bundle-Load-API.md

```diff
@@ -36,83 +36,48 @@ Given this it seems appropriate to have a new Bundle::LoadSymbols API for CppMic
 
 For the specific use case of DS (and similar extender clients), have an API in the CppMicroServices that
 
-1) loads shared library without dealing with OS specific data types
+1) loads shared library without exposing OS specific data types
 2) resolves symbols in a Bundle's shared library with the intent of calling exported functions
 3) does not require the users of this API to write any OS specific code e.g DS would only call bundle.GetSymbol() instead of using native OS calls
 
 ## Detailed design
 
 Bundle::GetSymbols() API will be declared in the file CppMicroServices/framework/include/cppmicroservices/Bundle.h as below
-The output of above API will be in the wrapper class which will be as belows.
+The output of above API will be in the function pointer type supplied by the API user.
 
 ```
-/**
-* @brief  This class holds the output of GetSymbol API.
-*
-* It acts as a wrapper over the handles returned by the OS APIs that loads the library.
-* Clients have to supply the function type as template argument to correctly get the desired symbol resolved using
-* GetSymbols function. See the usage workflow for more details.
-*
-*/
-
-namespace cppmicroservices {
-ls ..libmwsplashscreen.so
-class BundleHandle
-{
-    public:
-        BundleHandle():handle(nullptr) {}
-        ~BundleHandle() { handle = nullptr; }
-        BundleHandle(const BundleHandle&);
-        BundleHandle(BundleHandle&&);
-        BundleHandle& operator=(const BundleHandle&);
-        BundleHandle& operator=(BundleHandle&&);
-
-        template<typename T>
-        T ResolveSymbol(const std::string& symname);
-
-    private:
-        void * symHandle;
-        BundleHandle(void *handle) : handle(symHandle) {}
-        friend BundleHandle GetSymbols();
-}
-
-}
 
 /**
-* Retrieves the bundle symbol(s) using platform APIs and return a tuple of BundleHandle class object
+* Retrieves the bundle symbol(s) using platform APIs and returns a function pointer associated with the bundle object
 * It uses following OS APIs to achieve the same
 *
 * On POSIX   : dlopen/dlsym
 * On Windows : LoadLibraryW/GetProcAddress
 * 
```

Comment:
It should be documented as a post condition that the library associated with the bundle will be loaded into the process if the API did not throw.